backend/cmd/api/internal/model/questions.go

```diff
@@ -15,46 +14,34 @@ type Question struct {
 	QuestionID  int32     `json:"questionid"`
 	Question    string    `json:"question"`
 	NotesPrompt string    `json:"notesprompt"`
-	Order       int       `json:"order"`
+	Ordr        int       `json:"order"`
 	PillarID    int       `json:"pillarid"`
 	Pillar      *Pillar   `json:"pillar,omitempty"`
 	Function    *Function `json:"function,omitempty"`
 }
 
-func (q *Question) Save(ctx context.Context) error {
+func (q *Question) Save(ctx context.Context) (*Question, error) {
 
-	var (
-		sql       string
-		boundArgs []any
-		err       error
-	)
+	var sqlb SqlBuilder
 
 	if q.QuestionID == 0 {
-		sql, boundArgs, _ = sqlBuilder.
+		sqlb = stmntBuilder.
 			Insert("questions").
 			Columns(questionsColumns[1:]...).
-			Values(q.Question, q.NotesPrompt, q.Order, q.PillarID).
-			Suffix("RETURNING " + strings.Join(questionsColumns, ", ")).
-			ToSql()
+			Values(q.Question, q.NotesPrompt, q.Ordr, q.PillarID).
+			Suffix("RETURNING " + strings.Join(questionsColumns, ", "))
 	} else {
-		sql, boundArgs, _ = sqlBuilder.Update("questions").
+		sqlb = stmntBuilder.Update("questions").
 			Set("question", q.Question).
 			Set("notesprompt", q.NotesPrompt).
-			Set("ordr", q.Order).
+			Set("ordr", q.Ordr).
 			Set("pillarid", q.PillarID).
 			Where("questionid=?", q.QuestionID).
-			Suffix("RETURNING " + strings.Join(questionsColumns, ", ")).
-			ToSql()
+			Suffix("RETURNING " + strings.Join(questionsColumns, ", "))
 	}
 
-	row, err := queryRow(ctx, sql, boundArgs...)
-	if err != nil {
-		return trapError(err)
-	}
-
-	err = row.Scan(&q.QuestionID, &q.Question, &q.NotesPrompt, &q.Order, &q.PillarID)
+	return queryRow(ctx, sqlb, pgx.RowToStructByNameLax[Question])
```

Comment:
Most `queryRow` will be given `pgx.RowToStructByName` or the `Lax` version.
most `query` will be given `pgx.RowToAddrOfStructByName`

In rare cases a custom function is passed to account for specific details of the calling function that the generalized functions cannot address